web/src/lib/components/alert/alert.tsx

```diff
@@ -0,0 +1,49 @@
+import { h, Component } from 'preact';
+import Icon from '../icon';
+
+interface Props {
+  text: string;
+  active: boolean;
+}
+
+interface State {
+  active: boolean;
+}
+
+export default class Alert extends Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+
+    this.onClick = this.onClick.bind(this);
+  }
+
+  state: State = {
+    active: false,
+  };
+
+  componentWillReceiveProps(nextProps: Props) {
+    if (nextProps.active !== this.props.active) {
+      this.setState({
+        active: nextProps.active,
+      });
+    }
+  }
+
+  private onClick() {
+    this.setState({
+      active: false,
+    });
+  }
+
+  render() {
+    const className = 'alert ' + (this.state.active ? 'active' : '');
+
+    return (
+      <div className={className}>
+        {this.props.text}
+
+        <Icon type="x" onClick={this.onClick} className="icon" />
+      </div>
+    );
```

Comment:
@Gregoor so here the banner visibility is being switched by CSS, instead of not rendering it. Personally i'm used to doing it this way (i.e. with CSS).

Would you suggest not emitting any DOM nodes at all instead of emitting them and making them invisible? If so, is that a (P)React convention?

Reply:
It would be easy enough for me to change that to a conditional render, if that's what you want to do.

Reply:
Thanks @aerohitk, the component is looking good!

@nevik Regarding not emitting any html nodes: Yes, I'd still stand by that. Though before React 16 it was not possible to `return null` in `render()`, so not emitting any wasn't really possible. But in this case I'd indeed only return a `<div/>`, when it is `!active`.
Do you want to make that change @aerohitk?